pkg/registry/resolver_test.go

```diff
@@ -698,6 +698,57 @@ func TestResolve(t *testing.T) {
 		},
 		expectedErr:           errors.New(`test/test: workflow/ipi-aws: parameter "NOT_THE_STEP_ENV" is overridden in [test/test] but not declared in any step`),
 		expectedValidationErr: errors.New(`workflow/ipi-aws: parameter "NOT_THE_STEP_ENV" is overridden in [workflow/ipi-aws] but not declared in any step`),
+	}, {
+		name: "Workflow with trap command without grace_period",
+		config: api.MultiStageTestConfiguration{
+			ClusterProfile: api.ClusterProfileGCPCRIO,
+			Workflow:       &awsWorkflow,
+		},
+		workflowMap: WorkflowByName{
+			awsWorkflow: {
+				ClusterProfile: api.ClusterProfileAWS,
+				Pre: []api.TestStep{{
+					LiteralTestStep: &api.LiteralTestStep{
+						As:       "trapper-keeper",
+						From:     "installer",
+						Commands: "trap \"\" SIGINT",
+						Resources: api.ResourceRequirements{
+							Requests: api.ResourceList{"cpu": "1000m"},
+							Limits:   api.ResourceList{"memory": "2Gi"},
+						},
+						Environment: []api.StepParameter{
+							{Name: "STEP_ENV", Default: &stepEnv},
+						}},
+				}},
+			},
+		},
+		expectedErr: errors.New("test `trapper-keeper` has `commands` containing `trap` command, but test step is missing grace_period"),
+	}, {
+		name: "Workflow with best effort without timeout",
+		config: api.MultiStageTestConfiguration{
+			ClusterProfile: api.ClusterProfileGCPCRIO,
+			Workflow:       &awsWorkflow,
+		},
+		workflowMap: WorkflowByName{
+			awsWorkflow: {
+				ClusterProfile: api.ClusterProfileAWS,
+				Pre: []api.TestStep{{
+					LiteralTestStep: &api.LiteralTestStep{
+						As:         "best-effort",
+						From:       "installer",
+						Commands:   "openshift-cluster install",
+						BestEffort: &[]bool{true}[0],
+						Resources: api.ResourceRequirements{
+							Requests: api.ResourceList{"cpu": "1000m"},
+							Limits:   api.ResourceList{"memory": "2Gi"},
+						},
+						Environment: []api.StepParameter{
+							{Name: "STEP_ENV", Default: &stepEnv},
+						}},
+				}},
+			},
+		},
+		expectedErr: errors.New("test best-effort contains best_effort without timeout"),
```

Comment:
Can you also add positive tests, i,E. `No error when no trap command & not best effort`?